### ISSUE TITLE
missing NO ACTION strategy for mySql OnUpdateType

### ISSUE THREAD

**Issue by itavq** [typeorm/typeorm#2012]
**Issue type:**
[X] bug report

**Database system/driver:**
[X] `mssql`

**TypeORM version:**

[ ] `latest`

**Steps to reproduce or a small repository showing the problem:**

Hello,

I looked and found that 
https://github.com/typeorm/typeorm/blob/master/src/metadata/types/OnUpdateType.ts

is missing the NO ACTION option
(see https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html)
While one may use RESTRICT as alternative when generating tables others (such as myself) have a working DB schema with the NO ACTION notion.

Is it possible to add that?

Thanks

**Comment by pleerock** [comment]
Looks like only OnUpdateType missing NO ACTION, isnt it?

**Comment by itavq** [comment]
@pleerock you are right

**Comment by pleerock** [comment]
PRs are accepted

**Comment by itavq** [comment]
@pleerock 
The code change is trivial, but cannot get the tests to pass.

```
...
  database schema > column collation > mssql
    √ should correctly create column with collation option

  database schema > column collation > mysql
    1) should correctly create column with collation option


  38 passing (31s)
  3 pending
  1 failing

  1) database schema > column collation > mysql
       should correctly create column with collation option:
     TypeError: Cannot read property 'should' of undefined
      at Object.<anonymous> (test\functional\database-schema\column-collation\my                                             sql\column-collation-mysql.ts:34:51)
      at step (build\compiled\test\functional\database-schema\column-collation\m                                             ysql\column-collation-mysql.js:32:23)
      at Object.next (build\compiled\test\functional\database-schema\column-coll                                             ation\mysql\column-collation-mysql.js:13:53)
      at fulfilled (build\compiled\test\functional\database-schema\column-collat                                             ion\mysql\column-collation-mysql.js:4:58)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)
```

Could it be limited to specific MySql installation version? I'm running 5.7.17.

**Comment by pleerock** [comment]
'should' of undefined means value you are asserting is undefined. use `expect` instead

**Comment by itavq** [comment]
Thanks.
The afford mentioned error is of a local run of the existing tests with no modifications.
As I'm not the author of the test and considering it passes successfully elsewhere, I assume there's something else I'm missing here.

**Comment by pleerock** [comment]
ah thats right. Try to push your changes and submit a PR lets see what CI will say

**Comment by ShadowManu** [comment]
How are we on this? PostgreSQL also allows `ON UPDATE NO ACTION` but no luck yet.

**Comment by btxtiger** [comment]
Getting the similar issue when using [typeorm-model-generator](https://github.com/Kononnable/typeorm-model-generator) and `onUpdate` is filled with 'NO ACTION'
![image](https://user-images.githubusercontent.com/15804690/47018242-a4f86380-d154-11e8-96ec-a64ad6187687.png)

```ts
@ManyToOne(type => log_action, log_action => log_action.logs, {
    nullable: false,
    onDelete: 'NO ACTION',
    onUpdate: 'NO ACTION'
})
```

**Comment by AlexMesser** [comment]
this issue already fixed in #2823
